server/service/src/sensor/berlinger.rs

```diff
@@ -61,13 +62,24 @@ pub fn get_matching_sensor_breach(
     connection: &StorageConnection,
     sensor_id: &str,
     start_datetime: NaiveDateTime,
-    _end_datetime: NaiveDateTime,
+    _end_datetime: NaiveDateTime, // don't match on the end time, just the breach type and start time
     breach_type: &TemperatureBreachRowType,
 ) -> Result<Vec<TemperatureBreach>, RepositoryError> {
-    let filter = TemperatureBreachFilter::new()
+    let mut filter = TemperatureBreachFilter::new()
         .sensor(SensorFilter::new().id(EqualFilter::equal_to(sensor_id)))
-        .r#type(EqualFilter::equal_to_breach_type(&breach_type))
-        .start_datetime(DatetimeFilter::equal_to(start_datetime));
+        .r#type(EqualFilter::equal_to_breach_type(&breach_type));
+
+    match breach_type {
```

Comment:
The start time on its own isn't guaranteed to be unique - you could have a consecutive and cumulative breach starting at the same time

Reply:
But there is already a filter by breach_type, if it's filtered by both, wouldn't we find unique breach ?

Reply:
Yes,  start time + breach type should be unique (for a given sensor), but isn't that what this function is doing? The first bit is adding the sensor & breach type to the filter, and then the `match` is adding in the start time. The difference is that it's an exact match on the start time if it's a consecutive breach, but for cumulative breaches you can only have one per day so a start time any time during the same day is a match.

Reply:
As long as we never change the start time of a breach (which shouldn't happen if we're no longer trying to fix up slightly dodgy data), then the combination of start time and breach type (and sensor) should be unique. Thinking about time  zones makes my head hurt, but surely we're fine as long as all timestamps are adjusted so that any comparisons are all in the same time zone (or UTC or whatever)?

Reply:
>  but surely we're fine as long as all timestamps are adjusted so that any comparisons are all in the same time zone (or UTC or whatever)?

Yep, likewise, makes my head hurt too. 

Agree if all timestamps are adjusted it should be easier to think about